contrib/adaptive-compression/adapt.c

```diff
@@ -0,0 +1,591 @@
+#define DISPLAY(...) fprintf(stderr, __VA_ARGS__)
```

Comment:
You should put the `include`s first, and you need to add the copyright/license header.